lib/ruby_lsp/tapioca/addon.rb

```diff
@@ -92,14 +99,21 @@ def workspace_did_change_watched_files(changes)
           end
         end.compact
 
-        return if constants.empty?
+        return if constants.empty? && !has_route_change
 
         @rails_runner_client.trigger_reload
-        @rails_runner_client.delegate_notification(
-          server_addon_name: "Tapioca",
-          request_name: "dsl",
-          constants: constants,
-        )
+
+        if has_route_change
+          @rails_runner_client.delegate_notification(server_addon_name: "Tapioca", request_name: "route_dsl")
```

Comment:
How do the logs look with multiple forked processes running at the same time? Any issues of intertwined output?

Reply:
I don't think they would run at the same time because if the file is `routes.rb`, we skip looking for declarations in it. It's not very common for people to define modules or classes that need DSLs as part of routes.

If they were to run at the same time, then indeed the outputs would be mixed.

Reply:
The only possibility is if you switch branches, git pull, run `rubocop -a` or anything that will trigger multiple files to be updated all at once.

During normal options, like saving files manually, it should never happen.